implementation/core/src/test/java/io/smallrye/faulttolerance/core/bulkhead/CompletionStageThreadPoolBulkheadTest.java

```diff
@@ -226,42 +221,31 @@ public void shouldNotStartNextIfCSInProgress() throws Exception {
         party.organizer().waitForAll();
 
         CompletionStage<String> secondResult = bulkhead.apply(new InvocationContext<>(null));
-        waitUntilQueueSize(bulkhead, 1, 100);
+        waitUntilQueueSize(bulkhead, 1, 150);
```

Comment:
I guess you have to increase the timeout to 150 millis because Awaitility's default "probe" interval is 100 millis? I'd rather change the interval in the `await()` call chain. The hand-written probe loop used 50 millis, I'd be fine with using 10 or 20 I think.

Reply:
Actually, the hand-written probe used a timeout of 100 as argument and had a sleep(50) before each iteration. In the end, that could result in a timeout of 150. When I replaced with Awaitility and set a timeout of 100, a test started to fail. Then I increased to 150 and it passed.

Reply:
Ah yea, you're right, it could be up to 150 millis originally. However, the current situation is basically equivalent to checking if the condition holds, and if not, waiting 100 millis and checking again. Perhaps we could just fold and unconditionally wait 100 millis and be done with it? :-)

Reply:
I'm not sure if I got it. Do you mean revert from 150 to 100?

Reply:
No, I mean current code is equivalent to

```java
if (bulkhead.getQueueSize() != size) {
    Thread.sleep(100);
    if (bulkhead.getQueueSize() != size) {
        throw new SomeException();
    }
}
```

Maybe we can just do that directly, instead of pretending there's a wait loop.

I guess my preference would be to make the Awaitility interval shorter, 10 or 20 millis:

```java
await().pollInterval(10, TimeUnit.MILLISECONDS).atMost(...)...
```

And 100 millis should be enough of a timeout IMHO.

Reply:
Yes, changing the pool interval to 10 made the test pass with 100 millis (including on my machine :-) )